Frameworks/Foundation/NSException.mm

```diff
@@ -216,7 +244,18 @@ - (HRESULT)_hresult {
         return [hresultValue unsignedIntValue];
     }
 
-    TraceWarning(L"NSException", L"Called -hresult on an NSException without an HRESULT!");
-    return E_UNEXPECTED;
+    return [[self class] _HRESULTForExceptionName:self.name];
+}
+
+- (void)_processException {
+    HRESULT hr = [self _hresult];
+    HSTRING_HEADER header;
+    HSTRING hsReason = nullptr;
+    
+    if(self.reason != nil) {
+        LPCWSTR reason = (LPCWSTR)([[self reason] cStringUsingEncoding:NSUnicodeStringEncoding]);
+        WindowsCreateStringReference(reason, wcslen(reason), &header, &hsReason);
```

Comment:
Ah, now, I see (or at least I hope).  You *prefer* me to use Wrappers::HStringReference ?
If so, that is fine with me.

The original remark (from the original CR) was:
:-( Wrappers::HString*Reference*???
I did not know what you meant.    It is obvious to me now.
And then the original comment here:
You leak this. :-( (technically you don't....
Confused me even more because there was no leak.

For future reference, something like "prefer using Wrappers::HstringReference" would be more useful than ":-( Wrappers::HString*Reference*???"

I have no problems using HStringReference.  In retrospect, (again *painfully* obviously now) I don't know why I didn't use it in the first place.  Especially since I was pointed to it.  (Yeah, I'm feeling a little embarrassed.)


---
In reply to: [93301103](https://github.com/Microsoft/WinObjC/pull/1570#discussion_r93301103) [](ancestors = 93301103,93299684,93297284,93125423)